bokehjs/src/coffee/core/selection_manager.ts

```diff
@@ -10,47 +10,152 @@ import {DataSource} from "models/sources/data_source"
 export type Renderer = any
 export type RendererView = any
 
+export type HitTestResult = Selection | null
+
+export abstract class SelectionPolicy extends Model {
+
+  hit_test(geometry: Geometry, renderer_views: RendererView[]): HitTestResult{
```

Comment:
Given `SelectionPolicy` is an abstract class, this method would be better defined as abstract as well.